# 2 Wage equation

# 2.1 Модель 1

In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df=pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/wage2.csv')
df

wage  hours   IQ  KWW  educ  exper  tenure  age  married  black  south  \
0     769     40   93   35    12     11       2   31        1      0      0   
1     808     50  119   41    18     11      16   37        1      0      0   
2     825     40  108   46    14     11       9   33        1      0      0   
3     650     40   96   32    12     13       7   32        1      0      0   
4     562     40   74   27    11     14       5   34        1      0      0   
..    ...    ...  ...  ...   ...    ...     ...  ...      ...    ...    ...   
930   520     40   79   28    16      6       1   30        1      1      1   
931  1202     40  102   32    13     10       3   31        1      0      1   
932   538     45   77   22    12     12      10   28        1      1      1   
933   873     44  109   25    12     12      12   28        1      0      1   
934  1000     40  107   32    12     17      18   35        1      0      1   

     urban  sibs  brthord  meduc  feduc     lwage  
0        1     1      2.0    8.0    8.0  6.645091  
1        1     1      NaN   14.0   14.0  6.694562  
2        1     1      2.0   14.0   14.0  6.715384  
3        1     4      3.0   12.0   12.0  6.476973  
4        1    10      6.0    6.0   11.0  6.331502  
..     ...   ...      ...    ...    ...       ...  
930      0     0      1.0   11.0    NaN  6.253829  
931      1     7      7.0    8.0    6.0  7.091742  
932      0     9      NaN    7.0    NaN  6.287858  
933      0     1      1.0    NaN   11.0  6.771935  
934      0     5      1.0    NaN    NaN  6.907755  

[935 rows x 17 columns]

$$
log(wage) = \beta_0 + \beta_1 * age + \beta_2 * IQ
$$

In [3]:
# специфицируем модель через формулу
wage_eq1 = smf.ols(formula='np.log(wage)~age+IQ', data=df).fit()
# Коэфициенты модели с округлением
wage_eq1.params.round(3)

Intercept    5.077
age          0.024
IQ           0.009
dtype: float64

## Спецификация

$$
log(wage)=5.077+0.024*age+0.009*IQ
$$

1. При увеличении возраста на 1 год, зарплата увеличится на 2.4%
2. При увеличении  IQ на 1 пункт, зарплата увеличится на 0.9 %

# 2.2 Модель 2

$$
log(wage) = \beta_0 + \beta_1 * age + \beta_2 * IQ + \beta_3 * south + \beta_4 * urban + \beta_5 * married + \beta_6 * KWW
$$

In [6]:
# специфицируем модель через формулу
wage_eq1 = smf.ols(formula='np.log(wage)~age+IQ+south+urban+married+KWW', data=df).fit()
# Коэфициенты модели с округлением
wage_eq1.params.round(3)

Intercept    5.126
age          0.014
IQ           0.007
south       -0.101
urban        0.165
married      0.191
KWW          0.007
dtype: float64

$$
log(wage) = 5.126 + 0.014 * age + 0.007 * IQ -0.101 * south + 0.165 * urban + 0.191 * married + 0.007 * KWW
$$

1. При увеличении возраста на 1 год зарплата увеличивается на 1.4 %
2. При увеличении IQ на 1 пункт зарплата увеличивается на 0.7 % 

# 3 Output equation

In [7]:
df=pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

# 3.1 Модель 1

$$
log(output) = \beta_0 + \beta_1 * log(capital) + \beta_2 * log(labour)
$$

In [8]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept         -1.711
np.log(capital)    0.208
np.log(labour)     0.715
dtype: float64

$$
log(output) = -1.711 + 0.208 * log(capital) + 0.715 * log(labour)
$$

1. При увеличении капитала на 1% выпуск увеличивается на 0.208%
2. При увеличении числа сотрудников на 1% выпуск увеличивается на 0.715%

# 3.2 Модель 2

$$
log(output) = \beta_0 + \beta_1 * log(capital) + \beta_2 * log(labour) + \beta_3 * log(wage)
$$

In [11]:
# специфицируем модель через формулу
output_eq2 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)', data=df).fit()
# Коэфициенты модели с округление
output_eq2.params.round(3)

Intercept         -5.007
np.log(capital)    0.149
np.log(labour)     0.720
np.log(wage)       0.921
dtype: float64

$$
log(output) = -5.007 + 0.149 * log(capital) + 0.720 * log(labour) + 0.921 * log(wage)
$$

1. При увеличении капитала на 1% выпуск увеличивается на 0.149%
2. При увеличении числа сотрудников на 1% выпуск увеличивается на 0.720%

# 4 Cost equation

In [13]:
df=pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

# 4.1 Модель 1

$$
log(cost) = \beta_0 + \beta_1 * log(q)
$$

In [15]:
# специфицируем модель через формулу
cost_eq1 = smf.ols(formula='np.log(cost)~np.log(q)', data=df).fit()
# Коэфициенты модели с округлениe
cost_eq1.params.round(3)

Intercept   -3.841
np.log(q)    0.836
dtype: float64

$$
log(cost) = -3.841 + 0.836 * log(q)
$$

1. При увеличении общего выпуска электроэнергии на 1%, общие издержки за год увеличиваются на 0.836%

# 4.2 Модель 2

$$
log(сost) = \beta_0 + \beta_1 * log(q) + \beta_2 * log(pl) + \beta_3 * log(pf) + \beta_4 * log(pk)
$$

In [17]:
# специфицируем модель через формулу
cost_eq2 = smf.ols(formula='np.log(cost)~np.log(q)+np.log(pl)+np.log(pf)+np.log(pk)', data=df).fit()
# Коэфициенты модели с округлениe
cost_eq2.params.round(3)

Intercept    -7.472
np.log(q)     0.838
np.log(pl)    0.044
np.log(pf)    0.713
np.log(pk)    0.188
dtype: float64

$$
log(cost) = -7.472 + 0.838 * log(q) + 0.044 * log(pl) + 0.713 * log(pf) + 0.188 * log(pk)
$$

1. При увеличении общего выпуска электроэнергии на 1%, общие издержки за год увеличиваются на 0.838%